### PR TITLE
[REVIEW] Created .dockerignore file to prevent adding stale /cpp/build/* direc…

### PR DESCRIPTION
**Pull request by jdye64**
…tory from host machine into docker build environment which causes Cmake cache build errors

### CODE REVIEW

**Comment by randerzander** [comment]
run tests

**Comment by kkraus14** [comment]
Needs a changelog entry but otherwise looks good to me.

**Comment by kkraus14** [comment]
run tests

**Comment by harrism** [comment]
@jdye64 Could you please add a line to the CHANGELOG.md to satisfy the CI changelog check? Then we can get this merged into branch-0.5.

**Comment by jdye64** [comment]
Sure thing. Just added the change. Couldn't tell if this community preferred multiple commits to be merged by the contributor prior to approval or at merger time by the person merging. If you prefer for me to squash the commits just let me know.

**Comment by harrism** [comment]
run tests

File: CHANGELOG.md [hunk 1]
```diff
@@ -25,6 +25,7 @@
 - PR #454 Improve CSV reader docs and examples
 - PR #465 Added templated C++ API for RMM to avoid explicit cast to `void**`
 - PR #472 RMM: Created centralized rmm::device_vector alias and rmm::exec_policy
+- PR #617 Introduced .dockerignore file to project. Prevents stale cmake cache files from being "ADD" to the docker container if previously built on host machine which causes the container build to fail.
```
Comment (harrism) [inline CHANGELOG.md]: Shorter message is better here -- can always refer to GitHub PR for full details.

**Comment by harrism** [comment]
rerun tests

**Pull request merged**